### ISSUE TITLE
Trouble with some videos on abc.go.com

### ISSUE THREAD

**Issue by keybounce** [rg3/youtube-dl#13098]
Note that the season (series?) ending episodes are not downloading.

## Please follow the guide below

- You will be asked some questions and requested to provide some information, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *issue* (like that [x])
- Use *Preview* tab to see how your issue will actually look like

---

### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2017.05.14*. If it's not read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x ] I've **verified** and **I assure** that I'm running youtube-dl **2017.05.14**

### Before submitting an *issue* make sure you have:
- [ x] At least skimmed through [README](https://github.com/rg3/youtube-dl/blob/master/README.md) and **most notably** [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [ x] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

### What is the purpose of your *issue*?
- [ x] Bug report (encountered problems with youtube-dl)
- [ ] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

```
keybounceMBP:OnceUponATime michael$ youtube-dl -v --get-filename -a dl-list -i
[debug] System config: []
[debug] User config: ['-k', '-o', '%(series)s/s%(season_number)02d-e%(episode_number)02d-%(title)s.%(ext)s', '-f', '\nbest[ext=mp4][height>431][height<=576]/\nbestvideo[ext=mp4][height=480]+bestaudio[ext=m4a]/\nbest[ext=mp4][height>340][height<=431]/\nbestvideo[ext=mp4][height>360][height<=576]+bestaudio/\nbest[height>340][height<=576]/\nbestvideo[height>360][height<=576]+bestaudio/\nbestvideo[height=360]+bestaudio/\nbest[ext=mp4][height>=280][height<=360]/\nbest[height<=576]/\nworst', '--ap-mso', 'Dish', '--ap-username', 'PRIVATE', '--ap-password', 'PRIVATE', '--write-sub', '--write-auto-sub', '--sub-lang', 'en,enUS,en-us', '--sub-format', 'ass/srt/best', '--convert-subs', 'ass', '--embed-subs', '--recode-video', 'mp4', '--mark-watched']
[debug] Custom config: []
[debug] Command-line args: ['-v', '--get-filename', '-a', 'dl-list', '-i']
[debug] Batch file urls: ['http://abc.go.com/shows/once-upon-a-time/episode-guide/season-06/18-where-bluebirds-fly', 'http://abc.go.com/shows/once-upon-a-time/episode-guide/season-06/19-the-black-fairy', 'http://abc.go.com/shows/once-upon-a-time/episode-guide/season-06/20-the-song-in-your-heart', 'http://abc.go.com/shows/once-upon-a-time/episode-guide/season-06/21-the-final-battle-part-1', 'http://abc.go.com/shows/once-upon-a-time/episode-guide/season-06/22-the-final-battle-part-2', 'http://abc.go.com/shows/once-upon-a-time/episode-guide/season-06/199-the-final-battle-begins']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2017.05.14
[debug] Python version 3.6.1 - Darwin-13.4.0-x86_64-i386-64bit
[debug] exe versions: ffmpeg 3.2.4, ffprobe 3.2.4, rtmpdump 2.4
[debug] Proxy map: {}
WARNING: en subtitles not available for VDKA3840073
WARNING: enUS subtitles not available for VDKA3840073
Once Upon A Time/s06-e199-The Final Battle Begins.mp4
[debug] Using fake IP 3.28.123.238 (US) as X-Forwarded-For.
WARNING: en subtitles not available for VDKA3813168
WARNING: enUS subtitles not available for VDKA3813168
Once Upon A Time/s06-e19-The Black Fairy.mp4
WARNING: en subtitles not available for VDKA3827987
WARNING: enUS subtitles not available for VDKA3827987
Once Upon A Time/s06-e20-The Song in Your Heart.mp4
ERROR: Unable to extract post url; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/Users/michael/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 760, in extract_info
    ie_result = ie.extract(url)
  File "/Users/michael/bin/youtube-dl/youtube_dl/extractor/common.py", line 433, in extract
    ie_result = self._real_extract(url)
  File "/Users/michael/bin/youtube-dl/youtube_dl/extractor/go.py", line 119, in _real_extract
    url, video_id, requestor_id, resource)
  File "/Users/michael/bin/youtube-dl/youtube_dl/extractor/adobepass.py", line 1477, in _extract_mvpd_auth
    post_url = self._html_search_regex(r'<form id="syn_login_form"[^>]+action=(["\'])(?P<url>.+?)\1', provider_login_page, 'post url', group='url')
  File "/Users/michael/bin/youtube-dl/youtube_dl/extractor/common.py", line 791, in _html_search_regex
    res = self._search_regex(pattern, string, name, default, fatal, flags, group)
  File "/Users/michael/bin/youtube-dl/youtube_dl/extractor/common.py", line 782, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
youtube_dl.utils.RegexNotFoundError: Unable to extract post url; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

ERROR: Unable to extract post url; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/Users/michael/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 760, in extract_info
    ie_result = ie.extract(url)
  File "/Users/michael/bin/youtube-dl/youtube_dl/extractor/common.py", line 433, in extract
    ie_result = self._real_extract(url)
  File "/Users/michael/bin/youtube-dl/youtube_dl/extractor/go.py", line 119, in _real_extract
    url, video_id, requestor_id, resource)
  File "/Users/michael/bin/youtube-dl/youtube_dl/extractor/adobepass.py", line 1477, in _extract_mvpd_auth
    post_url = self._html_search_regex(r'<form id="syn_login_form"[^>]+action=(["\'])(?P<url>.+?)\1', provider_login_page, 'post url', group='url')
  File "/Users/michael/bin/youtube-dl/youtube_dl/extractor/common.py", line 791, in _html_search_regex
    res = self._search_regex(pattern, string, name, default, fatal, flags, group)
  File "/Users/michael/bin/youtube-dl/youtube_dl/extractor/common.py", line 782, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
youtube_dl.utils.RegexNotFoundError: Unable to extract post url; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

ERROR: Unable to extract post url; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "/Users/michael/bin/youtube-dl/youtube_dl/YoutubeDL.py", line 760, in extract_info
    ie_result = ie.extract(url)
  File "/Users/michael/bin/youtube-dl/youtube_dl/extractor/common.py", line 433, in extract
    ie_result = self._real_extract(url)
  File "/Users/michael/bin/youtube-dl/youtube_dl/extractor/go.py", line 119, in _real_extract
    url, video_id, requestor_id, resource)
  File "/Users/michael/bin/youtube-dl/youtube_dl/extractor/adobepass.py", line 1477, in _extract_mvpd_auth
    post_url = self._html_search_regex(r'<form id="syn_login_form"[^>]+action=(["\'])(?P<url>.+?)\1', provider_login_page, 'post url', group='url')
  File "/Users/michael/bin/youtube-dl/youtube_dl/extractor/common.py", line 791, in _html_search_regex
    res = self._search_regex(pattern, string, name, default, fatal, flags, group)
  File "/Users/michael/bin/youtube-dl/youtube_dl/extractor/common.py", line 782, in _search_regex
    raise RegexNotFoundError('Unable to extract %s' % _name)
youtube_dl.utils.RegexNotFoundError: Unable to extract post url; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.

keybounceMBP:OnceUponATime michael$ 
```

Note the following in that list:
1. ABC put up three episodes on 5/14, and episode 18 was expired early (apparently they only let you watch the last 5 episodes, not the last 5 weeks).
2. Attempting to fetch episode 18 gives the special (episode 199).
3. But fetching 199 directly, or either of the final 2 episodes, gives an error.

BUT:

If I only ask for the last two directly (not using a batch file):

```
keybounceMBP:OnceUponATime michael$ youtube-dl -v --get-filename http://abc.go.com/shows/once-upon-a-time/epis
ode-guide/season-06/21-the-final-battle-part-1
[debug] System config: []
[debug] User config: ['-k', '-o', '%(series)s/s%(season_number)02d-e%(episode_number)02d-%(title)s.%(ext)s', '-f', '\nbest[ext=mp4][height>431][height<=576]/\nbestvideo[ext=mp4][height=480]+bestaudio[ext=m4a]/\nbest[ext=mp4][height>340][height<=431]/\nbestvideo[ext=mp4][height>360][height<=576]+bestaudio/\nbest[height>340][height<=576]/\nbestvideo[height>360][height<=576]+bestaudio/\nbestvideo[height=360]+bestaudio/\nbest[ext=mp4][height>=280][height<=360]/\nbest[height<=576]/\nworst', '--ap-mso', 'Dish', '--ap-username', 'PRIVATE', '--ap-password', 'PRIVATE', '--write-sub', '--write-auto-sub', '--sub-lang', 'en,enUS,en-us', '--sub-format', 'ass/srt/best', '--convert-subs', 'ass', '--embed-subs', '--recode-video', 'mp4', '--mark-watched']
[debug] Custom config: []
[debug] Command-line args: ['-v', '--get-filename', 'http://abc.go.com/shows/once-upon-a-time/episode-guide/season-06/21-the-final-battle-part-1']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2017.05.14
[debug] Python version 3.6.1 - Darwin-13.4.0-x86_64-i386-64bit
[debug] exe versions: ffmpeg 3.2.4, ffprobe 3.2.4, rtmpdump 2.4
[debug] Proxy map: {}
WARNING: en subtitles not available for VDKA3840464
WARNING: enUS subtitles not available for VDKA3840464
Once Upon A Time/s06-e21-The Final Battle Part 1.mp4
keybounceMBP:OnceUponATime michael$ 
```

In other words, it is dying if I do a batch fetch, but working if I fetch one episode at a time.

**Comment by 89z** [comment]
Issue should be closed, those URL are all dead



http://abc.go.com/shows/once-upon-a-time/episode-guide/season-06/18-where-bluebirds-fly
http://abc.go.com/shows/once-upon-a-time/episode-guide/season-06/19-the-black-fairy
http://abc.go.com/shows/once-upon-a-time/episode-guide/season-06/20-the-song-in-your-heart
http://abc.go.com/shows/once-upon-a-time/episode-guide/season-06/21-the-final-battle-part-1
http://abc.go.com/shows/once-upon-a-time/episode-guide/season-06/22-the-final-battle-part-2
http://abc.go.com/shows/once-upon-a-time/episode-guide/season-06/199-the-final-battle-begins
